### PR TITLE
Rust testsuites

### PR DESCRIPTION
**Pull request by szg251**
- Implementation of `Prelude` and `Plutus` testsuites
- Fix a few issues:
  - adding `pub` keyword to prod and rec type fields
  - snake casing struct fields
- Adding `to_json_string` and `from_json_string` trait methods to Json

### CODE REVIEW

File: extras/lbf-nix/lbf-rust.nix [hunk 1]
```diff
@@ -78,13 +78,12 @@ let
...
     };
 
-  # 
   crateVersions = pkgs.writeTextFile {
     name = "lambda-buffers-crate-versions";
     text = ''
       num-bigint = "0.4.4"
       serde_json = { version = "1.0.107", features = ["arbitrary_precision"] }
-      plutus-ledger-api = { git = "https://github.com/mlabs-haskell/plutus-ledger-api-rust", features = [ "lbf", ], rev = "fb93fa590908580eb40368369bf6614d42ce9a95" }
+      plutus-ledger-api = { git = "https://github.com/mlabs-haskell/plutus-ledger-api-rust", features = [ "lbf", ], ref = "szg251/fix-lbf", rev = "4460f2ff8af7993a24209ebe28df5ef9b3b3b397" }
```
Comment (bladyjoker) [inline extras/lbf-nix/lbf-rust.nix:86]: It's going to be tricky managing updates across the board. Perhaps we should document and try to use tags for these foundational deps?

File: testsuites/lbt-plutus/lbt-plutus-rust/Cargo.toml [hunk 2]
```diff
@@ -4,10 +4,11 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
+plutus-ledger-api = { git = "https://github.com/mlabs-haskell/plutus-ledger-api-rust", features = [
+  "lbf",
+], ref = "szg251/fix-lbf", rev = "4460f2ff8af7993a24209ebe28df5ef9b3b3b397" }
```
Comment (bladyjoker) [inline testsuites/lbt-plutus/lbt-plutus-rust/Cargo.toml:9]: Yeah upgrades are going to be tricky.

Imagine the following!

Each `build.nix` file also declares a `packages.upgrade-script` which is actually a devShell in which `cargo update`, `npm i`, `spago2nix generate` and sh happens?

Hmm, curious if that could be smtn useful

**Review by bladyjoker — APPROVED** [review]
Woohoo!

File: extras/lbf-nix/lbf-rust.nix (hunk 1 above)
Comment (szg251) [reply extras/lbf-nix/lbf-rust.nix:86]: Yes, I want to improve this situation...

File: testsuites/lbt-plutus/lbt-plutus-rust/Cargo.toml (hunk 2 above)
Comment (szg251) [reply testsuites/lbt-plutus/lbt-plutus-rust/Cargo.toml:9]: I'll create an issue to solve the updating process. This needs some thinking

**Pull request merged**